### PR TITLE
17.0-technical_onboarding-basm

### PR DESCRIPTION
**Pull request by MohammedBasioni**

### CODE REVIEW

File: estate/models/estate_property.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')],
+        help="Orientation is used to separate different garden orientations")
+    
+
```
Comment (hupo-odoo) [inline estate/models/estate_property.py:24]: One empty line too much here

**Review by hupo-odoo — COMMENTED** [review]
Good work! I've made a few suggestions for improvement

File: estate/models/estate_property.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    bedrooms = fields.Integer()
+    living_area = fields.Integer()
+    facades = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[('north', 'North'), ('south', 'South'), ('east', 'East'), ('west', 'West')],
+        help="Orientation is used to separate different garden orientations")
```
[the remark this answers is not in the document]
Comment (hupo-odoo) [reply estate/models/estate_property.py:22]: you can also apply this comment to the other selection fields, makes things easier to read

File: estate/views/estate_property_views.xml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        </field>
+    </record>
+
+    <record id="real_estate_view_search" model="ir.ui.view">
+        <field name="name">real.estate.view.search</field>
+        <field name="model">estate.property</field>
+        <field name="arch" type="xml">
+            <search string="Search Properties">
+                <filter string="Postcode" name="groupby_postcode" context="{'group_by': 'postcode'}"/>
+                <filter string="Available" name="available" domain="['|', ('state', '=', 'new'), ('state', '=', 'recieved')]"/>
```
Comment (hupo-odoo) [inline estate/views/estate_property_views.xml:72]:
```suggestion
                <filter string="Available" name="available" domain="[('state', 'in', ('new', 'received'))]"/>
```

File: estate/models/estate_property.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    active = fields.Boolean(default=True)
+    state = fields.Selection(
+        default='new',
+        required=True,
+        copy=False,
+        selection=[
+            ('new', 'New'),
+            ('recieved', 'Offer Received'),
```
Comment (hupo-odoo) [inline estate/models/estate_property.py:30]:
```suggestion
            ('received', 'Offer Received'),
```

**Review by hupo-odoo — COMMENTED** [review]
Good work! :+1:  Don't hesitate to ask us on discord if you have any question. Don't get stuck too long on something

File: estate/models/estate_property.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (MohammedBasioni) [reply estate/models/estate_property.py]: Done! :))

File: estate/views/estate_property_views.xml (hunk 3 above)
Comment (MohammedBasioni) [reply estate/views/estate_property_views.xml]: Didn't know that I could do this! Thanks!

File: estate/models/estate_property.py (hunk 4 above)
Comment (MohammedBasioni) [reply estate/models/estate_property.py]: Good catch :)) Thanks

File: estate/views/estate_property_views.xml [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                    <t t-name="kanban-box">
+                        <div class="oe_kanban_global_click">
+                            <field name="name" />
+                            <div t-if="record.state.raw_value == 'new'">
+                                This is new!
+                            </div>
+                            <div>Expected Price: <field name="expected_price" /></div>
+
+                            <div t-if="record.state.raw_value == 'received'"> Best Price: <field
+                                    name="best_price" />
```
Comment (clbr-odoo) [inline estate/views/estate_property_views.xml:47]: That's some very strange indentation...

**Review by clbr-odoo — COMMENTED** [review]
Good job.
You still have some weird indentation habits, read some code from Odoo repository to get used to our ways :slightly_smiling_face: 
I'm only pointing errors/improvements to do once but there are multiple occurrences for some.

File: estate/models/estate_property_offer.py [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+    @api.model
+    def create(self, vals):
+        best_price = self.env['estate.property'].browse(
+            vals['property_id']).best_price
+        if best_price > vals['price']:
+            raise exceptions.UserError(
+                f"You can not make on offer less than the best price: {best_price}.")
+
+        self.env['estate.property'].browse(
+            vals['property_id']).state = 'received'
```
[the remark this answers is not in the document]
Comment (MohammedBasioni) [reply estate/models/estate_property_offer.py:74]: If I saved the value of the browse in a variable then I set the `state` in the second call will it reflect in the database automatically?

File: estate/models/estate_property.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+    @api.constrains('selling_price', 'expected_price')
+    def _check_selling_price(self):
+        for record in self:
+            if record.selling_price:
+                if float_utils.float_compare(record.selling_price, 0.9*record.expected_price, 2) < 0:
+                    raise exceptions.ValidationError(
+                        message="The selling price must be greater that 90% of the expected price!")
+
+    @api.ondelete(at_uninstall=False)
+    def _unlink_except_sold_or_canceled_property(self):
```
[the remark this answers is not in the document]
Comment (clbr-odoo) [reply estate/models/estate_property.py]: Well, it seems to me that it is either:
- allow unlink **only** `new` or `cancel` states
- allow unlink **except** `sold` state
You're mixing both :wink:

[the remark this answers is not in the document]
Comment (MohammedBasioni) [reply estate/models/estate_property.py]: I got it now 💯 I was thinking the opposite :)

**Pull request closed without merging**